In [2]:
import numpy as np
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [3]:
df = pd.DataFrame({
    'water': np.repeat(('daily', 'weekly'), 15),
    'sun': np.tile(np.repeat(('low', 'med', 'high'), 5), 2),
    'height': (6, 6, 6, 5, 6, 5, 5, 6, 4, 5, 
               6, 6, 7, 8, 7, 3, 4, 4, 4, 5,
               4, 4, 4, 4, 4, 5, 6, 6, 7, 8)
})

df

water   sun  height
0    daily   low       6
1    daily   low       6
2    daily   low       6
3    daily   low       5
4    daily   low       6
5    daily   med       5
6    daily   med       5
7    daily   med       6
8    daily   med       4
9    daily   med       5
10   daily  high       6
11   daily  high       6
12   daily  high       7
13   daily  high       8
14   daily  high       7
15  weekly   low       3
16  weekly   low       4
17  weekly   low       4
18  weekly   low       4
19  weekly   low       5
20  weekly   med       4
21  weekly   med       4
22  weekly   med       4
23  weekly   med       4
24  weekly   med       4
25  weekly  high       5
26  weekly  high       6
27  weekly  high       6
28  weekly  high       7
29  weekly  high       8

In [4]:
model = ols('height ~ C(water) + C(sun) + C(water):C(sun)', data=df).fit()
sm.stats.anova_lm(model, typ=2)

sum_sq    df        F    PR(>F)
C(water)          8.533333   1.0  16.0000  0.000527
C(sun)           24.866667   2.0  23.3125  0.000002
C(water):C(sun)   2.466667   2.0   2.3125  0.120667
Residual         12.800000  24.0      NaN       NaN

* Vì **p-value** của ***sun** và ***water*** đều nhỏ hơn $0.05$, điều này có nghĩa là cả hai yếu tố này đều có ý nghĩa về mặt thống kê ảnh hưởng đến ***height*** của cây.
* Và **p-value** của ***water:sun*** là $0.120667$, lớn hơn $0.05$, điều này có nghĩa là ***water*** và ***sun*** không có tương tác qua lại nào lẫn nhau.